Filter results quantity

jgoley opened the pull request:
Displays the total number of filtered/unfiltered rows in pagination and wraps come content in spans for styling flexibility.

aslagle commented on lib/reactive_table.js:
```diff
@@ -295,15 +296,21 @@ var getDefaultFieldVisibility = function (field) {
 }
 
+var getRowCount = function () {
+    if(this.isServer) {
```
this should be this.server - it always runs on the client and this.server indicates whether it's using a server-side publication.

jgoley commented:
@aslagle I've added tests total row count and updated the documentation, however, one of the reactivity tests (C: server-side collection) is failing and I'm not sure why.

jgoley commented:
Also: I wonder if "of" should be in a separate span from the number. My thought is that someone might want to style only the number.

aslagle commented:
The server-side collection test fails if you re-run it without restarting meteor - sorry about that.

aslagle commented on lib/reactive_table.html:
```diff
@@ -83,7 +83,14 @@
+                {{#if showRowCount}}
+                  <span class="total-row-count">{{i18n 'reactiveTable.of'}} {{showRowCount}}</span>
+                {{/if}}
+                <span class="rowsPerPage-label">{{i18n 'reactiveTable.rowsPerPage'}}</span>
```
All classes that aren't for bootstrap should start with reactive-table.

jgoley replied:
Should I change 'rows-per-page'?

aslagle replied:
Huh, I guess there are a few that don't follow it so these are okay if they're specific enough to the table. I do prefer dashes to camel case though. Don't change any that are already there in case people have customized them.

jgoley replied:
I'll change it so that it references the parent like so: ```.rows-per-page-count```

aslagle commented on lib/reactive_table.js:
```diff
@@ -217,7 +217,8 @@ var setup = function () {
     context.showNavigation = this.data.showNavigation || this.data.settings.showNavigation || 'always';
     context.showNavigationRowsPerPage = getDefaultTrueSetting('showNavigationRowsPerPage', this.data);
-    
+    context.showRowCount = this.data.showRowCount || this.data.settings.showRowCount || true;
```
I think the default should be false for backwards-compatibility, and you can probably use getDefaultFalseSetting

aslagle commented on lib/reactive_table.js:
```diff
@@ -295,15 +296,21 @@ var getDefaultFieldVisibility = function (field) {
 
+var getRowCount = function () {
+    if (this.server) {
+        return ReactiveTableCounts.findOne(this.publicationId.get());
```
this could do the (count ? count.count : 0) check and getPageCount wouldn't have to distinguish the server-side collections

aslagle commented on lib/reactive_table.js:
```diff
@@ -492,6 +498,14 @@ Template.reactiveTable.helpers({
+    },
+
+    'showRowCount': function() {
+        if(this.showRowCount)
```
I don't think you need to check this if it's already inside an if in the template

jgoley replied:
This is actually checking the settings since its optional.

aslagle replied:
Doesn't the template also check, so this would never get called if it's false?

jgoley replied:
The helper and property are named the same so I suppose the template is using what is returned from the helper.

aslagle replied:
Oh that's really confusing. Could the helper be 'getRowCount' instead?

jgoley replied:
Sure. Sorry about that.

aslagle commented on test/test_pagination.js:
```diff
@@ -59,6 +59,39 @@ testAsyncMulti('Pagination - previous/next controls', [function (test, expect) {
+
+Tinytest.add('Pagination - rows per page', function (test) {
+  var table = Blaze.renderWithData(
+    Template.testTmpl,
```
Why testTmpl instead of Template.reactiveTable? You can probably use my testTable helper if it doesn't need to be async.

aslagle commented:
I'm not sure it matters that much if 'of' is a separate span, but I don't mind it either.

aslagle commented on lib/reactive_table.js:
```diff
@@ -494,6 +497,12 @@ Template.reactiveTable.helpers({
+        var rowCount = getRowCount.call(this);
+        if (_.isObject(rowCount)) return rowCount.count;
+        else if (rowCount) return rowCount;
+        return '0';
```
I think you're already handling the object somewhere else, so this function could be removed and it could just be `'getRowCount': getRowCount`

aslagle commented on lib/reactive_table.html:
```diff
@@ -94,12 +102,17 @@
+                <i class="next-page fa fa-caret-right"></i>
               {{else}}
-                &nbsp;&nbsp;<label class="next-page">&gt;</label>
+                <label class="next-page">&gt;</label>
```
Does removing the `&nbsp;`s change the spacing of this section, if you haven't added any additional css?

jgoley replied:
As long as there is a new line in the html between spans there is space. The spacing seems the same. Should I add more?
I also removed ```&nbsp;``` from the page-number area. I added space to the next-page and previous-page and vertically centered the icon.

aslagle commented:
Released as version 0.8.2, thanks!
